tests/unit/_experimental/classification/test_utils.py

```diff
@@ -290,3 +290,152 @@ def test__compute_confusion_matrix__with_title() -> None:
         labels=["a", "not a"],
         matrix=[[1, 0], [1, 0]],
     )
+
+
+def test__roc_curve() -> None:
+    y_true = [1, 1, 0, 1, 0, 0, 1, 0, 1]
+    y_score = [0.3, 0.8, 0.5, 0.7, 0.1, 0.4, 0.9, 0.2, 0.6]
+    fpr, tpr = _roc_curve(y_true, y_score)
+    assert len(fpr) == len(tpr)
+    assert np.allclose(fpr, [0.0, 0.0, 0.0, 0.5, 0.5, 1.0])
+    assert np.allclose(tpr, [0.0, 0.2, 0.8, 0.8, 1.0, 1.0])
+
+
+def test__roc_curve__invalid() -> None:
+    # no negative samples
+    y_true = [1, 1, 1]
+    y_score = [0.1, 0.5, 0.6]
+    fpr, tpr = _roc_curve(y_true, y_score)
+    assert np.allclose(fpr, [])
+    assert np.allclose(tpr, [0.0, 1 / 3, 1.0])
+
+    # no positive samples
+    y_true = [0, 0, 0]
+    y_score = [0.1, 0.5, 0.6]
+    fpr, tpr = _roc_curve(y_true, y_score)
+    assert np.allclose(fpr, [0.0, 1 / 3, 1.0])
+    assert np.allclose(tpr, [])
+
+
+def test__compute_roc_curves__invalid() -> None:
```

Comment:
Nit: what's the reason `pytest.mark.parametrize` is not used for this test?

Reply:
hmm no particular reason.  It started off as a small test and more cases got added later.  Will use `parametrize`.